File: atcoder/abc395b_make_target.py
def create_grid(n):
    grid = [['#' for _ in range(n + 1)] for _ in range(n + 1)]
    
    for i in range(1, n + 1):
        j = n + 1 - i
        if i <= j:
            for k in range(i, j + 1):
                for l in range(i, j + 1):
                    grid[k][l] = '#' if i % 2 == 1 else '.'
    
    for i in range(1, n + 1):
        print("".join(grid[i][1:]))

if __name__ == "__main__":
    n = int(input())
    create_grid(n)

File: atcoder/abc395b_make_target.rs
use std::io;

fn create_grid(n: usize) {
    let mut grid = vec![vec!['#'; n + 1]; n + 1];
    
    for i in 1..=n {
        let j = n + 1 - i;
        if i <= j {
            for k in i..=j {
                for l in i..=j {
                    grid[k][l] = if i % 2 == 1 { '#' } else { '.' };
                }
            }
        }
    }
    
    for i in 1..=n {
        println!("{}", grid[i][1..].iter().collect::<String>());
    }
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();
    create_grid(n);
}
